context.go

```diff
@@ -41,18 +45,19 @@ func (c *Context) Response() http.ContextResponse {
 	return NewContextResponse(c.instance, &BodyWriter{ResponseWriter: c.instance.Writer})
 }
 
-func (c *Context) WithValue(key string, value any) {
-	c.instance.Set(key, value)
+func (c *Context) WithValue(key any, value any) {
+	c.instance.Set(fmt.Sprintf("%v", key), value) // need to store the key-val to the underlying gin too
```

Comment:
If `c.Ctx` exists, what's the reason for still using `c.instance.Set`? Is there any risk if we don't set it?

Reply:
it failed the test in `group_test` @hwbrzzl

Reply:
<img width="676" alt="Screenshot 2024-09-13 at 16 39 48" src="https://github.com/user-attachments/assets/6c7a16ba-ee07-4ed0-940c-aaa8673a6431">

Reply:
What's the specific error?

Reply:
<img width="844" alt="Screenshot 2024-09-13 at 17 05 17" src="https://github.com/user-attachments/assets/652479a0-5039-474f-babd-b9564daf00e5">

Reply:
Thanks, I'll check it.

Reply:
I checked this situation, the current code can not pass test cases when setting value in a global middleware, the value will be lost in [26 line](https://github.com/goravel/gin/pull/88/files#diff-552f47512a00afe5fc6850cc9ddc830a6daeca162750e50aab4ed549685e0253R26). We need to set a value into the gin context.

The context of gin only supports `map[string]any`, it can't be used directly, but we can make a hack:

<img width="570" alt="image" src="https://github.com/user-attachments/assets/5cc62491-d1db-492e-8a7a-c9251b7113e1">

Then, the `Value` method should be optimized, too. This is just an idea, you can try it.  

And please add a test case to test the struct key situation.

<img width="1006" alt="image" src="https://github.com/user-attachments/assets/73fb1278-d76a-4758-b788-58024d928b12">

Reply:
i see, i think it may work, let me try. thanks @hwbrzzl

Reply:
wonderful, it pass the tests, please check again @hwbrzzl